packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -1067,10 +1101,38 @@ public void resumePreview() {
   public void startPreview() throws CameraAccessException {
     if (pictureImageReader == null || pictureImageReader.getSurface() == null) return;
     Log.i(TAG, "startPreview");
-
     createCaptureSession(CameraDevice.TEMPLATE_PREVIEW, pictureImageReader.getSurface());
   }
 
+  private void startPreviewWithVideoRendererStream()
+      throws CameraAccessException, InterruptedException {
+    if (videoRenderer == null) return;
+
+    // get rotation for rendered video
+    final PlatformChannel.DeviceOrientation lockedOrientation =
+        ((SensorOrientationFeature) cameraFeatures.getSensorOrientation())
+            .getLockedCaptureOrientation();
+    int rotation =
+        lockedOrientation == null
+            ? cameraFeatures
+                .getSensorOrientation()
+                .getDeviceOrientationManager()
+                .getVideoOrientation()
+            : cameraFeatures
+                .getSensorOrientation()
+                .getDeviceOrientationManager()
+                .getVideoOrientation(lockedOrientation);
+
+    if (cameraProperties.getLensFacing() != initialCameraFacing) {
```

Comment:
It may also be good to test that the rotation is set correctly depending on the case here.